content/influxdb/cloud-dedicated/reference/client-libraries/v3/python.md

```diff
@@ -7,18 +7,490 @@ menu:
+
+## API reference
+
+The `influxdb_client_3` module includes the following classes and functions.
+
+<!-- TOC -->
+
+- [Classes](#classes)
+- [Functions](#functions)
+- [Constants](#constants)
+- [Exceptions](#exceptions)
+
+## Classes
+
+- [Class InfluxDBClient3](#class-influxdbclient3)
+  - [InfluxDBClient3.write](#influxdbclient3write)
+  - [InfluxDBClient3.write_file](#influxdbclient3write_file)
+  - [InfluxDBClient3.query](#influxdbclient3query)
+  - [InfluxDBClient3.close](#influxdbclient3close)
+- [Class Point](#class-point)
+- [Class WriteOptions](#class-writeoptions)
+
+## Class InfluxDBClient3
+
+Provides an interface for interacting with InfluxDB APIs for writing and querying data.
+
+### Syntax
+
+```py
+__init__(self, host=None, org=None, database=None, token=None,
+        _write_client_options=None, _flight_client_options=None, **kwargs)
+```
+
+Initializes and returns an `InfluxDBClient3` instance with the following:
+
+- A singleton _write client_ configured for writing to the database.
+- A singleton _Flight client_ configured for querying the database.
+
+### Attributes
+
+- **org** (str): The organization name (for {{% cloud-name %}}, set this to an empty string (`""`)).
+- **database** (str): The database to use for writing and querying.
+- **_write_client_options** (dict): Options to use when writing to InfluxDB.
+  If `None`, writes are [synchronous](#synchronous-writing).
+- **_flight_client_options** (dict): Options to use when querying InfluxDB.
+
+#### Batch writing
+
+In batching mode, the client adds the record or records to a batch, and then schedules the batch for writing to InfluxDB.
+The client writes the batch to InfluxDB after reaching `_write_client_options.batch_size` or `_write_client_options.flush_interval`.
+If a write fails, the client reschedules the write according to the `_write_client_options` retry options.
+
+To use batching mode, pass an instance of `WriteOptions` for the `InfluxDBClient3._write_client_options` argument--for example:
+
+1.  Instantiate `WriteOptions()` with defaults or with
+`WriteOptions.write_type=WriteType.batching`.
+
+    ```py
+      # Batching with all batch size, flush, and retry defaults
+      write_options = WriteOptions()
+    ```
+
+2.  Call the a `write_client_options` function to create an options object that uses `write_options` from the preceding step.
+
+    ```py
+    wco = write_client_options(WriteOptions=write_options)
+    ```
+
+3. Initialize the client, setting the `_write_client_options` argument to the `wco` object from the preceding step.
+
+    {{< tabs-wrapper >}}
+{{% code-placeholders "DATABASE_(NAME|TOKEN)" %}}
+```py
+with InfluxDBClient3(token="DATABASE_TOKEN", host="cluster-id.influxdb.io",
+                    org="", database="DATABASE_NAME",
+                    _write_client_options=wco) as client:
+
+    client.write(record=points)
+```
+{{% /code-placeholders %}}
+    {{< /tabs-wrapper >}}
```

Comment:
No tabs here.
~~~suggestion

    {{% code-placeholders "DATABASE_(NAME|TOKEN)" %}}
```py
with InfluxDBClient3(token="DATABASE_TOKEN", host="cluster-id.influxdb.io",
                    org="", database="DATABASE_NAME",
                    _write_client_options=wco) as client:

    client.write(record=points)
```
    {{% /code-placeholders %}}

~~~

Reply:
> No tabs here.

I did this b/c `code-placeholders` breaks when indented. I thought `tabs-wrapper` was a workaround at the time, but I'm not sure about that now.